pkg/kv/kvserver/rafttrace/rafttrace.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package rafttrace
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverpb"
+	"github.com/cockroachdb/cockroach/pkg/raft"
+	"github.com/cockroachdb/cockroach/pkg/raft/raftpb"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing/tracingpb"
+	"github.com/cockroachdb/logtags"
+)
+
+type traceValue struct {
+	traceID tracingpb.TraceID
+	spanID  tracingpb.SpanID
+	// ctx is a trace specific context used to log events on this trace.
+	ctx context.Context
+	// baseCtx is the underlying proposal buffer which we additionally log to.
+	baseCtx withContext
+}
+
+// logf logs the message to the trace context and the base context. The base
+// context is populated on the leaseholder and is attached to the SQL trace.
+func (t traceValue) logf(depth int, format string, args ...interface{}) {
+	log.InfofDepth(t.ctx, depth+1, format, args...)
+	if t.baseCtx != nil {
+		if ctx := t.baseCtx.Context(); ctx != nil {
+			log.VEventfDepth(ctx, depth+1, 3, format, args...)
+		}
+	}
+}
+
+func tracingID(traceID tracingpb.TraceID, spanID tracingpb.SpanID) string {
+	// We attempt to balance uniqueness with readability by only keeping the
+	// lower 16 bits of the trace and span.
```

Comment:
It ends up being 8 chars that you are searching for (4 from trace and 4 from span). I meant to post the logs before but will do that now. I don't think we will have duplicates in practice since this is 2^32 possible pairs.